docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,428 @@
+    --policy file://${DIR_CCO}/oidc-bucket-policy.json
+```
+
+Now the CloudFront Distribution must have access to the Bucket object `/pvt/keys.json`, test it:
+
+```
+$ curl https://${CLOUDFRONT_HOST}/keys.json
+```
+
+### Create the new OIDC using CloudFront Distribution<a name="setup-oidc-idp"></a>
+
+- Extract the service account signer public key, to generate the IdP by `ccoctl`:
+
+```bash
+oc get configmap bound-sa-token-signing-certs \
+    --namespace openshift-kube-apiserver \
+    --output json \
+    | jq --raw-output '.data["service-account-001.pub"]' \
+    > serviceaccount-signer.public
+```
+
+- Generate the IdP files into the local directory `new-oidc`:
+
+```bash
+./ccoctl aws create-identity-provider \
+    --name=${CLUSTER_NAME} \
+    --region=${CLUSTER_REGION} \
+    --public-key-file=${PWD}/serviceaccount-signer.public \
+    --output-dir=new-oidc/ \
+    --dry-run
+```
+
+- Patch the IdP OIDC to the new Domain name
+
+```bash
+sed -i "s/${OIDC_BUCKET_HOST}/${CLOUDFRONT_HOST}/g" new-oidc/04-iam-identity-provider.json
+```
+
+- Discover the thumbprint for the keys from the CloudFront Distribution URL:
+
+> AWS Docs - Getting the Thumbprint: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+
+```bash
+openssl s_client -servername ${CLOUDFRONT_HOST} \
+    -showcerts -connect ${CLOUDFRONT_HOST}:443 </dev/null \
+    | openssl x509 -outform pem > certificate.crt
+
+export CERT_THUMBPRINT=$(openssl x509 -in certificate.crt -fingerprint -sha1 -noout \
+    | awk -F'=' '{print$2}' | tr -d ':')
+
+jq -r ".ThumbprintList=[\"$CERT_THUMBPRINT\"]" ${PWD}/new-oidc/04-iam-identity-provider.json \
+    > ${PWD}/new-oidc/04-iam-identity-provider-new.json
+```
+
+- Create the identity provider AWS OIDC:
+
+```bash
+aws iam create-open-id-connect-provider \
+    --cli-input-json file://${PWD}/new-oidc/04-iam-identity-provider-new.json \
+    > ${PWD}/new-oidc//04-iam-identity-provider-object.json 
+
+# Describe or get from file
+#export OIDC_ARN=$(aws iam list-open-id-connect-providers |jq -r ".OpenIDConnectProviderList[] | select(.Arn | endswith(\"${CLOUDFRONT_URI}\")).Arn")
+
+export OIDC_ARN=$(jq -r .OpenIDConnectProviderArn ${PWD}/new-oidc//04-iam-identity-provider-object.json)
+
+echo ${OIDC_ARN}
+```
+
+## Backup existing state <a name="backup"></a>
+
+- Get Objects and existing tests
+
+```bash
+export CURRENT_PATH=$PWD/current-cluster
+mkdir $CURRENT_PATH
+oc get authentication -o yaml |tee -a $CURRENT_PATH/authentication.yaml
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}" \
+    | jq -r '.Credentials=""' \
+    | tee ${CURRENT_PATH}/identities.json
+```
+
+- Save the current IAM Roles
+
+```bash
+aws iam list-roles \
+    | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-openshift\"))" \
+    | tee ${CURRENT_PATH}/iam-roles.json
```

Comment:
I encountered the issue here.
When I executed this command, no result found.
My 'CLUSTER_NAME' is jshu-connected and the created role name is like jshu-connected-**21746**-openshift-cloud-credential-operator-cloud-c.
So I revised the command as below and it worked 
[cloud-user@preserve-for-hive-test testdir]$ aws iam list-roles | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-**21746**-openshift\"))" | tee ${CURRENT_PATH}/iam-roles.json

Reply:
As I comment on Slack, I think it is quite related to your environment and how that var is populated when creating the ccoctl. btw I changed the strategy of that vars, discovering it based on tags.
Thanks for reporting it.